src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -392,45 +329,81 @@ EmberAfNodeOperationalCertStatus ConvertToNOCResponseStatus(CHIP_ERROR err)
 bool emberAfOperationalCredentialsClusterRemoveAllFabricsCallback(EndpointId endpoint, app::CommandHandler * commandObj)
 {
     emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Remove all Fabrics");
-    PlatformMgr().ScheduleWork(DoRemoveAllFabrics, 0);
     emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_SUCCESS);
     return true;
 }
 
 bool emberAfOperationalCredentialsClusterAddNOCCallback(EndpointId endpoint, app::CommandHandler * commandObj, ByteSpan NOCArray,
-                                                        ByteSpan IPKValue, NodeId CaseAdminNode, uint16_t AdminVendorId)
+                                                        ByteSpan IPKValue, NodeId adminNodeId, uint16_t adminVendorId)
 {
     EmberAfNodeOperationalCertStatus nocResponse = EMBER_ZCL_NODE_OPERATIONAL_CERT_STATUS_SUCCESS;
 
-    CHIP_ERROR err      = CHIP_NO_ERROR;
-    uint8_t fabricIndex = 0;
+    CHIP_ERROR err          = CHIP_NO_ERROR;
+    FabricIndex fabricIndex = 0;
 
     emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: commissioner has added an Op Cert");
 
+    err = gFabricBeingCommissioned.SetOperationalCertsFromCertArray(NOCArray);
+    VerifyOrExit(err == CHIP_NO_ERROR, nocResponse = ConvertToNOCResponseStatus(err));
+
+    gFabricBeingCommissioned.SetVendorId(adminVendorId);
+
+    err = GetGlobalFabricTable().AddNewFabric(gFabricBeingCommissioned, fabricIndex);
+    VerifyOrExit(err == CHIP_NO_ERROR, nocResponse = ConvertToNOCResponseStatus(err));
+
+    err = GetGlobalFabricTable().Store(fabricIndex);
+    VerifyOrExit(err == CHIP_NO_ERROR, nocResponse = ConvertToNOCResponseStatus(err));
+
+    // We have a new operational identity and should start advertising it.  We
+    // can't just wait until we get network configuration commands, because we
+    // might be on the operational network already, in which case we are
+    // expected to be live with our new identity at this point.
+    chip::app::Mdns::AdvertiseOperational();
```

Comment:
This line really makes a good point that:

1. We should rename `chip::app::Mdns` to `chip::app::dnssd` (since Thread doesn't use mDNS!)
2. This is a good example for now, but having this a blocking call may be problematic. How long does it take to return from this? Do we have any guarantees? Feels like the application should be given a chance to organize things as it sees fit.

Reply:
> How long does it take to return from this? 

I was assuming this was a pretty async process so would not block on any network activity, but @andreilitvin should confirm.

Reply:
Issue to track the first suggestion: https://github.com/project-chip/connectedhomeip/issues/8908

Reply:
for the second question: https://github.com/project-chip/connectedhomeip/issues/8909

Reply:
I believe we control only our 'own' implementation, which involves setting up some memory data (what we advertise) and (re-)creating some UDP listening sockets, potentially also an intial 'send network everywhere'.

That being said, pure mdns spec I believe would have 'detect colliding names' and conflict resolution which minmdns does not do specifically (with the potentially unsound logic that using nodeid/macaddress/somethingelseunique is good enough). Generic mDNS can take longer.

Would it be fair to require 'this is non-blocking and if anything requires blocking (like detecting conflicting names) that will happen async in the background?'z

Reply:
From this code's point of view, that would be ideal.